sdk/core/core-util/test/public/aborterUtils.spec.ts

```diff
@@ -0,0 +1,154 @@
+  const createPromise = ({
+    abortSignal,
+    abortErrorMsg,
+  }: { abortSignal?: AbortSignalLike; abortErrorMsg?: string } = {}): Promise<unknown> =>
+    createAbortablePromise(
+      (resolve) => {
+        token = setTimeout(resolve, delayTime);
+      },
+      {
+        cleanupBeforeAbort: () => clearTimeout(token),
+        abortSignal,
+        abortErrorMsg,
+      }
+    );
+  afterEach(function () {
+    sinon.restore();
+  });
+
+  it("should resolve if not aborted nor rejected", async function () {
+    const clock = sinon.useFakeTimers();
+    const promise = createPromise();
+    const time = await clock.nextAsync();
+    clock.restore();
+    assert.strictEqual(time, delayTime);
+    await assert.isFulfilled(promise);
+  });
+
+  it("should reject when aborted", async function () {
+    const aborter = new AbortController();
+    const abortErrorMsg = "The test operation was aborted.";
+    const promise = createPromise({
+      abortSignal: aborter.signal,
+      abortErrorMsg,
+    });
+    aborter.abort();
+    await assert.isRejected(promise, abortErrorMsg);
+  });
+});
+
+describe("cancelablePromiseRace", function () {
+  let function1Aborted = false;
+  let function2Aborted = false;
+  let function3Done = false;
+  const function1Delay = 100;
+  let function2Delay = 200;
+  const function3Delay = 2000; // Default: function1Delay < function2Delay < function3Delay
+  const function2Message = "function 2 is rejected";
+
+  const function1 = async (abortOptions: { abortSignal?: AbortSignalLike }): Promise<number> => {
+    let token: ReturnType<typeof setTimeout>;
+    return createAbortablePromise(
+      (resolve) => {
+        token = setTimeout(resolve, function1Delay);
+      },
+      {
+        cleanupBeforeAbort: () => {
+          function1Aborted = true;
+          clearTimeout(token);
+        },
+        abortSignal: abortOptions.abortSignal,
+      }
+    );
+  };
+
+  const function2 = async (abortOptions: { abortSignal?: AbortSignalLike }): Promise<string> => {
+    let token: ReturnType<typeof setTimeout>;
+    return createAbortablePromise(
+      (reject) => {
+        token = setTimeout(() => reject(function2Message), function2Delay);
+      },
+      {
+        cleanupBeforeAbort: () => {
+          function2Aborted = true;
+          clearTimeout(token);
+        },
+        abortSignal: abortOptions.abortSignal,
+      }
+    );
+  };
+
+  const function3 = async (abortOptions: { abortSignal?: AbortSignalLike }): Promise<void> => {
+    return delay(function3Delay, { abortSignal: abortOptions.abortSignal }).finally(() => {
+      function3Done = true;
+    });
+  };
+
+  afterEach(function () {
+    // reset to default values
+    function1Aborted = false;
+    function2Aborted = false;
+    function2Delay = 200;
+    function3Done = false;
+  });
+
+  it("should resolve with the first promise that resolves, abort the rest", async function () {
+    const startTime = Date.now();
```

Comment:
I'm a little wary of doing math with clocktime in tests unless we're mocking the clock to begin with.

Reply:
remove the delay function for an in-place function with my trackers in it :)
and got rid of the clocks and math, let me know if that looks good